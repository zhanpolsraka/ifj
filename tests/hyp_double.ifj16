//0
class Main{
	static double a = 3;
	static double b = 4;
	static void run(){
		double hypotenuse = hyp();
	}
	static double hyp(){
		double aa = Main.a*Main.a;
		double bb = Main.b*Main.b;
		double c = aa+bb;
		return c;
	}
}
