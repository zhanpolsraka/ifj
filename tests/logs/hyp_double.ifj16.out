---------------FILENAME: tests/hyp_double.ifj16---------------


Instruction stack item [24] -> class with name "Main"
Instruction stack item [23] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [22] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [21] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [20] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [19] -> function with name "run"
Instruction stack item [18] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [17] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [16] -> end function
Instruction stack item [15] -> function with name "hyp"
Instruction stack item [14] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 3 -> MUL
Instruction stack item [11] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [10] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [9] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [8] -> instruction with operation 3 -> MUL
Instruction stack item [7] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [4] -> instruction with operation 1 -> PLUS
Instruction stack item [3] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class


---------------MAKE RELATIONS--------------
                     *
                     *
                     *
------------------EXECUTE------------------

------------------OUTPUT-------------------

STEP 1 - non-func instructions
Found Main
STEP 1 - non-func instructions
INSERT 3
---------------------------------------------------
N0: type is 22, values is 3
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 3 to 0x76f400
---------------------------------------------------
N0: type is 22, values is 3
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
INSERT 4
---------------------------------------------------
N0: type is 22, values is 4
---------------------------------------------------
STEP 1 - non-func instructions
ASSIGN 4 to 0x76f4c0
---------------------------------------------------
N0: type is 22, values is 4
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
Found run()
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
Got out of run()
STEP 1 - non-func instructions
Got in function - ignore instructions until I get out
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
STEP 1 - non-func instructions
Got out of function
STEP 1 - non-func instructions
Got out of Main
STEP 2 - validating existance of run() in Main
STEP 3 - executing run() instructions
INSTR_CALL_FUNC
INSERT 3
---------------------------------------------------
N0: type is 23, values is 3
---------------------------------------------------
INSERT 3
---------------------------------------------------
N1: type is 23, values is 3
N0: type is 23, values is 3
---------------------------------------------------
UNKNOWN INSTRUCTION
---------------------------------------------------
N0: type is 23, values is 9
---------------------------------------------------
ASSIGN 9.000000 to 0x76fa70
---------------------------------------------------
N0: type is 23, values is 9
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 4
---------------------------------------------------
N0: type is 23, values is 4
---------------------------------------------------
INSERT 4
---------------------------------------------------
N1: type is 23, values is 4
N0: type is 23, values is 4
---------------------------------------------------
UNKNOWN INSTRUCTION
---------------------------------------------------
N0: type is 23, values is 16
---------------------------------------------------
ASSIGN 16.000000 to 0x76f260
---------------------------------------------------
N0: type is 23, values is 16
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 9
---------------------------------------------------
N0: type is 23, values is 9
---------------------------------------------------
INSERT 16
---------------------------------------------------
N1: type is 23, values is 16
N0: type is 23, values is 9
---------------------------------------------------
PLUS
---------------------------------------------------
N0: type is 23, values is 25
---------------------------------------------------
ASSIGN 25.000000 to 0x7703b0
---------------------------------------------------
N0: type is 23, values is 25
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------
INSERT 25
---------------------------------------------------
N0: type is 23, values is 25
---------------------------------------------------
---------------------------------------------------
N0: type is 23, values is 25
---------------------------------------------------
ASSIGN 25.000000 to 0x76f860
---------------------------------------------------
N0: type is 23, values is 25
---------------------------------------------------
---------------------------------------------------
---------------------------------------------------

-----------------END OUTPUT----------------


----------------ALL RIGHT!!!---------------

----------------SYMBOL TREE----------------

Class Main
	Variable -> a, type DOUBLE, value [3.000000]
	Variable -> b, type DOUBLE, value [4.000000]
	Function -> run
Lokal variables: 
	Variable -> hypotenuse, type DOUBLE, value [25.000000]
	Function -> hyp
Lokal variables: 
	Variable -> aa, type DOUBLE, value [9.000000]
	Variable -> bb, type DOUBLE, value [16.000000]
	Variable -> c, type DOUBLE, value [25.000000]


--------------INSTRUCTION LIST-------------

Instruction stack item [24] -> class with name "Main"
Instruction stack item [23] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [22] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [21] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [20] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [19] -> function with name "run"
Instruction stack item [18] -> instruction with operation 35 -> INSTR_CALL_FUNC
Instruction stack item [17] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [16] -> end function
Instruction stack item [15] -> function with name "hyp"
Instruction stack item [14] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [13] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [12] -> instruction with operation 3 -> MUL
Instruction stack item [11] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [10] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [9] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [8] -> instruction with operation 3 -> MUL
Instruction stack item [7] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [6] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [5] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [4] -> instruction with operation 1 -> PLUS
Instruction stack item [3] -> instruction with operation 28 -> ASSIGNMENT
Instruction stack item [2] -> instruction with operation 37 -> INSTR_ASS_ARG
Instruction stack item [1] -> end function
Instruction stack item [0] -> end class



